include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -296,8 +300,29 @@
 #define _ONEDPL_CPP20_SHIFT_LEFT_RIGHT_PRESENT                                                                         \
     (_ONEDPL___cplusplus >= 202002L && (_MSC_VER >= 1921 || _GLIBCXX_RELEASE >= 10))
 
-#define _ONEDPL_CPP20_CONCEPTS_PRESENT                                                                                 \
-    (_ONEDPL___cplusplus >= 202002L && (_MSC_VER >= 1923 || _GLIBCXX_RELEASE >= 10 || _LIBCPP_VERSION >= 13000))
+#if _ONEDPL___cplusplus >= 202002L && __cpp_concepts >= 201907L && __cpp_lib_concepts >= 202002L
+#    define _ONEDPL_CPP20_CONCEPTS_PRESENT 1
+#else
+#    define _ONEDPL_CPP20_CONCEPTS_PRESENT 0
+#endif
+
+#if _ONEDPL___cplusplus >= 202302L && __cpp_lib_tuple_like >= 202207L
+#    define _ONEDPL_CPP23_TUPLE_LIKE_COMMON_REFERENCE_PRESENT 1
+#else
+#    define _ONEDPL_CPP23_TUPLE_LIKE_COMMON_REFERENCE_PRESENT 0
+#endif
+
+#if _ONEDPL___cplusplus >= 202302L && __cpp_lib_ranges_zip >= 202110L
+#    define _ONEDPL_CPP23_RANGES_ZIP_PRESENT 1
+#else
+#    define _ONEDPL_CPP23_RANGES_ZIP_PRESENT 0
+#endif
+
+// To use std::tuple as a proxy reference and satisfy iterator concepts for C++20 and beyond, we need
+// the changes to std::tuple in P2321R2 and the tuple-like basic_common_reference specialization in P2165R4.
```

Comment:
```
// When C++20 concepts are available, we must use std::tuple as a proxy reference to satisfy iterator concepts, which // requires the changes to std::tuple in P2321R2 and the tuple-like basic_common_reference specialization in P2165R4.
```
In my opinion, this or some similar phrasing makes the part of the expression before the `||` more clear.

Reply:
I agree this is clearer and have added this suggestion.